grammar.js

```diff
@@ -78,12 +78,8 @@ module.exports = grammar({
     // Simple statements
 
     _simple_statements: $ => seq(
-      $._simple_statement,
-      optional(repeat(seq(
-        $._semicolon,
-        $._simple_statement
-      ))),
-      optional($._semicolon),
+      sep1($._simple_statement, ';'),
+      optional(';'),
```

Comment:
I think this isn't exactly the same as the before, it doesn't cover the first case, only the second, and it allows the third.

```python
a = 'a'; b = 'b''
a = 'a'; b = 'b';
a = 'a'; b = 'b';;  # this is wrong
```

Would be great to add a test for this.

Reply:
Added tests in https://github.com/tree-sitter/tree-sitter-python/pull/135/commits/29b828eba2e201b563229e6c27ebbb9e0a56be88. They are passing locally.

Reply:
I think the tests are passing because it's being parsed as a whole, if you parse every expression individually you get the error

```
~/github/tree-sitter-python on  semicolon! ⌚ 13:45:27
$ cat test.py
   1   a = 'a'; b = 'b''

~/github/tree-sitter-python on  semicolon! ⌚ 13:45:29
$ npm run parse -- test.py

> tree-sitter-python@0.19.0 parse
> tree-sitter parse "test.py"

(module [0, 0] - [1, 0]
  (ERROR [0, 0] - [0, 17]
    (expression_statement [0, 0] - [0, 7]
      (assignment [0, 0] - [0, 7]
        left: (identifier [0, 0] - [0, 1])
        right: (string [0, 4] - [0, 7])))
    (identifier [0, 9] - [0, 10])
    (string [0, 13] - [0, 16])))
test.py 0 ms    (ERROR [0, 0] - [0, 17])

~/github/tree-sitter-python on  semicolon! ⌚ 13:45:32
$ cat test.py
   1   a = 'a'; b = 'b';

~/github/tree-sitter-python on  semicolon! ⌚ 13:45:40
$ npm run parse -- test.py

> tree-sitter-python@0.19.0 parse
> tree-sitter parse "test.py"

(module [0, 0] - [1, 0]
  (expression_statement [0, 0] - [0, 7]
    (assignment [0, 0] - [0, 7]
      left: (identifier [0, 0] - [0, 1])
      right: (string [0, 4] - [0, 7])))
  (expression_statement [0, 9] - [0, 16]
    (assignment [0, 9] - [0, 16]
      left: (identifier [0, 9] - [0, 10])
      right: (string [0, 13] - [0, 16]))))

~/github/tree-sitter-python on  semicolon! ⌚ 13:45:42
$ cat test.py
   1   a = 'a'; b = 'b';;  # this is wrong

~/github/tree-sitter-python on  semicolon! ⌚ 13:45:51
$ npm run parse -- test.py

> tree-sitter-python@0.19.0 parse
> tree-sitter parse "test.py"

(module [0, 0] - [1, 0]
  (expression_statement [0, 0] - [0, 7]
    (assignment [0, 0] - [0, 7]
      left: (identifier [0, 0] - [0, 1])
      right: (string [0, 4] - [0, 7])))
  (expression_statement [0, 9] - [0, 16]
    (assignment [0, 9] - [0, 16]
      left: (identifier [0, 9] - [0, 10])
      right: (string [0, 13] - [0, 16])))
  (expression_statement [0, 17] - [0, 17]
    (identifier [0, 17] - [0, 17]))
  (comment [0, 20] - [0, 35]))
test.py 0 ms    (MISSING identifier [0, 17] - [0, 17])

```

Reply:
With master, the last case results in

```
(module [0, 0] - [1, 0]
  (expression_statement [0, 0] - [0, 7]
    (assignment [0, 0] - [0, 7]
      left: (identifier [0, 0] - [0, 1])
      right: (string [0, 4] - [0, 7])))
  (ERROR [0, 7] - [0, 17]
    (expression_statement [0, 9] - [0, 16]
      (assignment [0, 9] - [0, 16]
        left: (identifier [0, 9] - [0, 10])
        right: (string [0, 13] - [0, 16])))))
test.py 0 ms    (ERROR [0, 7] - [0, 17])

```

Reply:
I've tried reverting to how the rule is in master but it did not result in an error. Might have something to do with the semicolon now being a literal instead of a `$ => *` rule.

Reply:
Never mind, I think this is fine (it generates a missing node)